autoload/go/guru.vim

```diff
@@ -180,78 +122,35 @@ function! s:async_guru(args) abort
   endif
 
   let state = {
-        \ 'status_dir': expand('%:p:h'),
-        \ 'statusline_type': printf("%s", a:args.mode),
         \ 'mode': a:args.mode,
-        \ 'status': {},
-        \ 'exitval': 0,
-        \ 'closed': 0,
-        \ 'exited': 0,
-        \ 'messages': [],
         \ 'parse' : get(a:args, 'custom_parse', funcref("s:parse_guru_output"))
       \ }
 
-  function! s:callback(chan, msg) dict
-    call add(self.messages, a:msg)
-  endfunction
-
-  function! s:exit_cb(job, exitval) dict
-    let self.exited = 1
-
-    let status = {
-          \ 'desc': 'last status',
-          \ 'type': self.statusline_type,
-          \ 'state': "finished",
-          \ }
-
-    if a:exitval
-      let self.exitval = a:exitval
-      let status.state = "failed"
-    endif
-
-    call go#statusline#Update(self.status_dir, status)
-
-    if self.closed
-      call self.complete()
-    endif
-  endfunction
-
-  function! s:close_cb(ch) dict
-    let self.closed = 1
-
-    if self.exited
-      call self.complete()
-    endif
-  endfunction
-
-  function state.complete() dict
-    let out = join(self.messages, "\n")
-
-    call self.parse(self.exitval, out, self.mode)
+  function! s:complete(job, exit_status, messages) dict abort
+    let output = join(a:messages, "\n")
+    call self.parse(a:exit_status, output, self.mode)
   endfunction
+  " explicitly bind complete to state so that within it, self will
+  " always refer to state. See :help Partial for more information.
+  let state.complete = function('s:complete', [], state)
+
+  let opts = {
+        \ 'status_type': printf("%s", a:args.mode),
+        \ 'for': '_',
+        \ 'errorformat': "%f:%l.%c-%[%^:]%#:\ %m,%f:%l:%c:\ %m",
+        \ 'complete': state.complete,
+        \ }
 
-  " explicitly bind the callbacks to state so that self within them always
-  " refers to state. See :help Partial for more information.
-  let start_options = {
-        \ 'callback': function('s:callback', [], state),
-        \ 'exit_cb': function('s:exit_cb', [], state),
-        \ 'close_cb': function('s:close_cb', [], state)
-       \ }
+  let opts = go#job#Options(l:opts)
 
   if has_key(result, 'stdin_content')
     let l:tmpname = tempname()
     call writefile(split(result.stdin_content, "\n"), l:tmpname, "b")
-    let l:start_options.in_io = "file"
-    let l:start_options.in_name = l:tmpname
+    let l:opts.in_io = "file"
+    let l:opts.in_name = l:tmpname
   endif
 
-  call go#statusline#Update(state.status_dir, {
-        \ 'desc': "current status",
-        \ 'type': state.statusline_type,
-        \ 'state': "analysing",
-        \})
```

Comment:
Is this removed accidentally? This is a core part of my workflow because otherwise, I don't have any idea what's going on. Same when it's finished (Above it's also removed)

Reply:
It's removed on purpose. Our job's api supports statusline updates. But, I'd misspelled the key. That's been fixed, and I see the status updates now. PTAL.

Reply:
Just tested again and it works fine 👍